playwright-e2e/tests/dsm/kitUploadFlow/saliva-kit-upload-flow.spec.ts

```diff
@@ -0,0 +1,180 @@
+import {test} from '@playwright/test';
+import {WelcomePage} from 'pages/dsm/welcome-page';
+import HomePage from 'pages/dsm/home-page';
+import {Navigation} from 'lib/component/dsm/navigation/navigation';
+import {login} from 'authentication/auth-dsm';
+import {StudyEnum} from 'lib/component/dsm/navigation/enums/selectStudyNav-enum';
+import ParticipantListPage from 'pages/dsm/participantList-page';
+import {StudyNavEnum} from 'lib/component/dsm/navigation/enums/studyNav-enum';
+import ParticipantPage from 'pages/dsm/participant-page/participant-page';
+import {KitUploadInfo} from 'pages/dsm/kitUpload-page/models/kitUpload-model';
+import ContactInformationTab from 'lib/component/dsm/tabs/contactInformationTab';
+import {TabEnum} from 'lib/component/dsm/tabs/enums/tab-enum';
+import {SamplesNavEnum} from 'lib/component/dsm/navigation/enums/samplesNav-enum';
+import {KitTypeEnum} from 'lib/component/dsm/kitType/enums/kitType-enum';
+import KitUploadPage from 'pages/dsm/kitUpload-page/kitUpload-page';
+import InitialScanPage from 'pages/dsm/initialScan-page';
+import FinalScanPage from 'pages/dsm/finalScan-page';
+import crypto from 'crypto';
+import SampleInformationTab from 'lib/component/dsm/tabs/sampleInformationTab';
+import {SampleInfoEnum} from 'lib/component/dsm/tabs/enums/sampleInfo-enum';
+import {SampleStatusEnum} from 'lib/component/dsm/tabs/enums/sampleStatus-enum';
+import KitsWithoutLabelPage from 'pages/dsm/kitsInfo-pages/kitsWithoutLabel-page';
+import {KitsColumnsEnum} from 'pages/dsm/kitsInfo-pages/enums/kitsColumns-enum';
+import KitsSentPage from 'pages/dsm/kitsInfo-pages/kitsSentPage';
+import KitsReceivedPage from 'pages/dsm/kitsInfo-pages/kitsReceived-page/kitsReceivedPage';
+
+
+test.describe.parallel('Saliva Kits upload flow', () => {
+  let welcomePage: WelcomePage;
+  let homePage: HomePage;
+  let navigation: Navigation;
+
+  let shortID: string;
+  let kitUploadInfo: KitUploadInfo;
+  let kitLabel: string;
+  let shippingID: string;
+
+  const studies = [StudyEnum.LMS, StudyEnum.OSTEO2];
+  const kitType = KitTypeEnum.SALIVA;
+  const expectedKitTypes = [KitTypeEnum.SALIVA, KitTypeEnum.BLOOD];
+
+  test.beforeEach(async ({ page, request }) => {
+    await login(page);
+    welcomePage = new WelcomePage(page);
+    homePage = new HomePage(page);
+    navigation = new Navigation(page, request);
+  });
+
+  for (const study of studies) {
+    test(`Should upload a single kit for one participant @functional @visual @dsm @${study}`, async () => {
+      await welcomePage.selectStudy(study);
+      await homePage.assertWelcomeTitle();
+      await homePage.assertSelectedStudyTitle(study);
+
+      const participantListPage = await navigation.selectFromStudy<ParticipantListPage>(StudyNavEnum.PARTICIPANT_LIST);
+      await participantListPage.assertPageTitle();
+
+      // find the participant that has not more than 14 samples
```

Comment:
What happens when no participant matches this selection criteria? Why need this?

Reply:
to avoid the "too many kits" error. 
If that happens, we are in trouble :D but it should not as these tests will run along with participant enrollment tests and at some point, they will have new participants. 

Note: I am planning to add pagination, in case we need to search for the participant on the other page.